base/linalg/triangular.jl

```diff
@@ -39,6 +39,9 @@ for t in (:LowerTriangular, :UnitLowerTriangular, :UpperTriangular,
     end
 end
 
+LowerTriangular(U::UpperTriangular) = throw(ArgumentError("this shouldn't happen. Please submit a bug report"))
+UpperTriangular(U::LowerTriangular) = throw(ArgumentError("this shouldn't happen. Please submit a bug report"))
```

Comment:
I don't think this is quite right, a user could easily call the constructors to try to do this. Maybe something like "Cannot create an UpperTriangular matrix from a LowerTriangular input"?